scanomatic/io/jsonizer.py

```diff
@@ -0,0 +1,178 @@
+import json
+from enum import Enum, unique
+import logging
+from typing import Any, Type, Union
+from collections.abc import Callable
+import numpy as np
+
+from scanomatic.generics.model import Model
+from scanomatic.models.factories.analysis_factories import (
+    AnalysisModelFactory,
+    GridModelFactory,
+)
+from scanomatic.models.analysis_model import COMPARTMENTS, VALUES, MEASURES
+from scanomatic.models.factories.rpc_job_factory import RPC_Job_Model_Factory
+from scanomatic.models.rpc_job_models import JOB_STATUS, JOB_TYPE
+
+
+class JSONSerializationError(ValueError):
+    pass
+
+
+class JSONDecodingError(JSONSerializationError):
+    pass
+
+
+class JSONEncodingError(JSONSerializationError):
+    pass
+
+
+CONTENT = "__CONTENT__"
+
+
+MODEL_CLASSES: dict[str, Callable[..., Model]] = {
+    "AnalysisModel": AnalysisModelFactory.create,
+    "GridModel": GridModelFactory.create,
+    "RPCjobModel": RPC_Job_Model_Factory.create,
+}
+
+
+def decode_model(obj: dict) -> Model:
+    encoding = SOMSerializers.MODEL.encoding
+    try:
+        creator = MODEL_CLASSES[obj[encoding]]
+    except KeyError:
+        msg = f"'{obj.get(encoding)}' is not a recognized model"
+        logging.error(msg)
+        raise JSONDecodingError(msg)
+    try:
+        content: dict = obj[CONTENT]
+    except KeyError:
+        msg = f"Serialized model {obj[encoding]} didn't have any content"
+        logging.error(msg)
+        raise JSONDecodingError(msg)
+
+    try:
+        return creator(**{
+            k: object_hook(v) if isinstance(v, dict) else v
+            for k, v in content.items()
+        })
+    except (TypeError, AttributeError):
+        msg = f"Serialized model {obj[encoding]} didn't contain a dict as content: {content}"  # noqa: E501
+        logging.error(msg)
+        raise JSONDecodingError(msg)
+
+
+ENUM_CLASSES: dict[str, Type[Enum]] = {
+    "COMPARTMENTS": COMPARTMENTS,
+    "VALUES": VALUES,
+    "MEASURES": MEASURES,
+    "JOB_TYPE": JOB_TYPE,
+    "JOB_STATUS": JOB_STATUS,
+}
+
+
+def decode_enum(obj: dict) -> Enum:
+    encoding = SOMSerializers.ENUM.encoding
+    try:
+        e = ENUM_CLASSES[obj[encoding]]
+    except KeyError:
+        msg = f"'{obj.get(encoding)}' is not a recognized enum"
+        logging.error(msg)
+        raise JSONDecodingError(msg)
+    content = obj.get(CONTENT)
+    if not isinstance(content, str):
+        msg = f"'{content}' is not one of the allowed string values fro {type(e).__name__}"  # noqa: E501
```

Comment:
```suggestion
        msg = f"'{content}' is not one of the allowed string values for {type(e).__name__}"  # noqa: E501
```
(inne i review så kan inte comitta direkt)